chord_drs/utils.py

```diff
@@ -14,3 +18,44 @@ def drs_file_checksum(path: str) -> str:
             hash_obj.update(chunk)
 
     return hash_obj.hexdigest()
+
+
+def _iter_over_async(async_generator: AsyncGenerator):
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+    iterator = async_generator.__aiter__()
+
+    async def get_next() -> tuple[bool, Any]:
+        try:
+            obj = await iterator.__anext__()
+            return False, obj
+        except StopAsyncIteration:
+            return True, None
+        except Exception as e:
+            current_app.logger.error(e)
```

Comment:
use logger.exception instead, which should auto-add exception info: https://docs.python.org/3/library/logging.html#logging.Logger.exception